tests/conftest.py

```diff
@@ -52,6 +53,26 @@ def eager_task_loop_factory() -> asyncio.AbstractEventLoop:
     )
 
 
+@pytest.fixture(autouse=True)
+def blockbuster() -> Iterator[BlockBuster]:
+    with blockbuster_ctx(
+        "anyio", excluded_modules=["anyio.pytest_plugin", "anyio._backends._asyncio"]
+    ) as bb:
+        bb.functions["os.read"].can_block_in(
+            "asyncio/base_events.py", "subprocess_exec"
+        )
+        for func in ["os.stat", "os.unlink"]:
+            bb.functions[func].can_block_in(
+                "anyio/_core/_sockets.py", "setup_unix_local_socket"
```

Comment:
`setup_unix_local_socket` is the only coroutine making blocking calls to `os.stat` and `os.unlink` .
Interestingly, asyncio's `create_unix_server` does exactly the same !
I did raise an issue for asyncio : https://github.com/python/cpython/issues/129807 but that was handled as `won't fix` by Guido himself 😄 . I let you decide if you want to do something about it. Blockbuster is about detecting potential issues, not forcing anything.

Reply:
While I usually like being pedantic about these things, I feel that in 99.999% of cases blocking will be a non-issue, and using threads will degrade performance. So add the exemptions please.